framework/src/parser/Parser.C

```diff
@@ -1996,6 +2004,19 @@ Parser::setScalarParameter<PostprocessorName, PostprocessorName>(
   }
 }
 
+template <>
+void
+Parser::setScalarParameter<ReporterName, std::string>(
+    const std::string & full_name,
+    const std::string & /*short_name*/,
+    InputParameters::Parameter<ReporterName> * param,
+    bool /*in_global*/,
+    GlobalParamsAction * /*global_block*/)
+{
+  std::vector<std::string> names = MooseUtils::split(_root->param<std::string>(full_name), "::");
+  param->set() = ReporterName(names[0], names[1]);
+}
```

Comment:
This doesn't seem to be documented anywhere. Basically to set a reporter name input parameter you have `object_name::value_name`. This confused me at first because the resulting combined name is `object_name/value_name`. I think this will be confusing for others too because the CSV and console output of a reporter is `object_name/value_name`. Ideally, I think we should choose one or the other. But I can easily be convinced that this is impossible.

Reply:
Also there needs to be an error if `::` isn't in the name. There's a segfault when `::` isn't used.

Reply:
I changed this to a / and added some doco